addons/website_quote/models/order.py

```diff
@@ -1,190 +1,167 @@
+    @api.multi
+    def _compute_amount_undiscounted(self):
+        for order in self:
             total = 0.0
             for line in order.order_line:
                 total += (line.product_uom_qty * line.price_unit)
-            res[order.id] = total
-        return res
-
-    _columns = {
-        'access_token': fields.char('Security Token', required=True, copy=False),
-        'template_id': fields.many2one('sale.quote.template', 'Quotation Template', readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}),
-        'website_description': fields.html('Description'),
-        'options' : fields.one2many('sale.order.option', 'order_id', 'Optional Products Lines', readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, copy=True),
-        'amount_undiscounted': fields.function(_get_total, string='Amount Before Discount', type="float",
-            digits_compute=dp.get_precision('Account')),
-        'quote_viewed': fields.boolean('Quotation Viewed')
-    }
-
-    def _get_template_id(self, cr, uid, context=None):
+            self.amount_undiscounted = total
+
+    @api.model
+    def _default_template_id(self):
         try:
-            template_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'website_quote', 'website_quote_template_default')[1]
+            res = self.env['ir.model.data'].xmlid_to_res_id('website_quote.website_quote_template_default')
+            quote_template = self.env['sale.quote.template'].search([('id', '=', res)])
         except ValueError:
-            template_id = False
-        return template_id
-
-    _defaults = {
-        'access_token': lambda self, cr, uid, ctx={}: str(uuid.uuid4()),
-        'template_id' : _get_template_id,
-    }
-
-    def open_quotation(self, cr, uid, quote_id, context=None):
-        quote = self.browse(cr, uid, quote_id[0], context=context)
-        self.write(cr, uid, quote_id[0], {'quote_viewed': True}, context=context)
+            quote_template = self.template_id
+        return quote_template
+
+    access_token = fields.Char(string='Security Token', required=True, copy=False, default=lambda self: str(uuid.uuid4()))
+    template_id = fields.Many2one('sale.quote.template', string='Quotation Template', default=_default_template_id, readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]})
+    website_description = fields.Html(string='Description')
+    options = fields.One2many('sale.order.option', 'order_id', string='Optional Products Lines', readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, copy=True)
+    amount_undiscounted = fields.Float(compute="_compute_amount_undiscounted", string='Amount Before Discount', digits_compute=dp.get_precision('Account'))
+    quote_viewed = fields.Boolean(string='Quotation Viewed')
+
+    @api.multi
+    def open_quotation(self):
+        self.quote_viewed = True
```

Comment:
add a `self.ensure_one()`, and prefix this method with `action_` (don't forget to change it in the view).